tests/main/preseed/task.yaml

```diff
@@ -42,51 +43,104 @@ execute: |
+    MATCH "Done .+ copy-snap-data +Copy snap \"core18\" data" < tasks.log
+    MATCH "Done .+ setup-profiles +Setup snap \"core18\" \([0-9]+\) security profiles" < tasks.log
+    MATCH "Done .+ link-snap +Make snap \"core18\" \([0-9]+\) available to the system" < tasks.log
+    MATCH "Done .+ auto-connect +Automatically connect eligible plugs and slots of snap \"core18\"" < tasks.log
+    MATCH "Done .+ set-auto-aliases +Set automatic aliases for snap \"core18\"" < tasks.log
+    MATCH "Done .+ setup-aliases +Setup snap \"core18\" aliases" < tasks.log
+
+    MATCH "Done .+ prerequisites +Ensure prerequisites for \"lxd\" are available" < tasks.log
+    MATCH "Done .+ prepare-snap +Prepare snap \"/var/lib/snapd/seed/snaps/lxd_[0-9]+.snap\" \([0-9]+\)" < tasks.log
+    MATCH "Done .+ mount-snap +Mount snap \"lxd\" \([0-9]+\)" < tasks.log
+    MATCH "Done .+ copy-snap-data +Copy snap \"lxd\" data" < tasks.log
+    MATCH "Done .+ setup-profiles +Setup snap \"lxd\" \([0-9]+\) security profiles" < tasks.log
+    MATCH "Done .+ link-snap +Make snap \"lxd\" \([0-9]+\) available to the system" < tasks.log
+    MATCH "Done .+ auto-connect +Automatically connect eligible plugs and slots of snap \"lxd\"" < tasks.log
+    MATCH "Done .+ connect +Connect lxd:network to snapd:network" < tasks.log
+    MATCH "Done .+ connect +Connect lxd:lxd-support to snapd:lxd-support" < tasks.log
+    MATCH "Done .+ connect +Connect lxd:system-observe to snapd:system-observe" < tasks.log
+    MATCH "Done .+ connect +Connect lxd:network-bind to snapd:network-bind" < tasks.log
+    MATCH "Done .+ setup-profiles +Setup snap \"lxd\" \([0-9]+\) security profiles for auto-connections" < tasks.log
+    MATCH "Done .+ set-auto-aliases +Set automatic aliases for snap \"lxd\"" < tasks.log
+    MATCH "Done .+ setup-aliases +Setup snap \"lxd\" aliases" < tasks.log
+
+    echo "Checking that there were no other 'Done' tasks when preseeding"
+    [ "$(grep -c ' Done ' tasks.log)" = "32" ]
+
+    # everything below is pending execution on first boot
+    MATCH "Do .+ run-hook +Run install hook of \"core18\" snap if present" < tasks.log
+    MATCH "Do .+ start-snap-services +Start snap \"lxd\" \([0-9]+\) services" < tasks.log
+    MATCH "Do .+ run-hook +Run configure hook of \"lxd\" snap if present" < tasks.log
+    MATCH "Do .+ run-hook +Run health check of \"lxd\" snap" < tasks.log
+    MATCH "Do .+ mark-seeded +Mark system seeded" < tasks.log
+    MATCH "Do .+ run-hook +Run install hook of \"snapd\" snap if present" < tasks.log
+    MATCH "Do .+ start-snap-services +Start snap \"snapd\" \([0-9]+\) services" < tasks.log
+    MATCH "Do .+ run-hook +Run configure hook of \"core\" snap if present" < tasks.log
+    MATCH "Do .+ start-snap-services  +Start snap \"core18\" \([0-9]+\) services" < tasks.log
+    MATCH "Do .+ run-hook +Run health check of \"core18\" snap" < tasks.log
+    MATCH "Do .+ run-hook +Run install hook of \"lxd\" snap if present" < tasks.log
+  else
```

Comment:
nitpick: I wonder if we should be explicit here that this applies to 19.10 and have an else that just fails if we run it on something unexpected?

Reply:
Good idea, will do.